scanner/indexer/service.go

```diff
@@ -1,12 +1,44 @@
 package indexer
 
-// TODO: Make this into a gRPC service.
-type Service struct {
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type indexerService struct {
+	v4.UnimplementedIndexerServer
 	indexer *Indexer
 }
 
-func NewService(indexer *Indexer) (*Service, error) {
-	return &Service{
+func NewIndexerService(indexer *Indexer) (*indexerService, error) {
+	return &indexerService{
 		indexer: indexer,
 	}, nil
 }
+
+func (s *indexerService) CreateIndexReport(ctx context.Context, req *v4.CreateIndexReportRequest) (*v4.IndexReport, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method CreateIndexReport not implemented")
+}
+func (s *indexerService) GetIndexReport(ctx context.Context, req *v4.GetIndexReportRequest) (*v4.IndexReport, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method GetIndexReport not implemented")
+}
+func (s *indexerService) HasIndexReport(ctx context.Context, req *v4.HasIndexReportRequest) (*types.Empty, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method HasIndexReport not implemented")
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *indexerService) RegisterServiceServer(grpcServer *grpc.Server) {
+	v4.RegisterIndexerServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
```

Comment:
Is this needed since we don't use the gateway at this time?

Reply:
It's part of the interface.